onelogin/get.go

```diff
@@ -216,3 +198,45 @@ func Get(app, provider string) (*awsprovider.Credentials, error) {
 
 	return &creds, nil
 }
+
+func getDevice(devices GenerateSamlAssertionResponseDevices) (deviceID, deviceType string, err error) {
+	// we might have more than one device
+	if len(devices) > 1 {
+		var selection int
+		for {
+			for i, d := range devices {
+				fmt.Printf("%d. %d - %s\n", i+1, d.DeviceId, d.DeviceType)
+			}
+
+			fmt.Printf("Please choose an MFA device to authenticate with (1-%d): ", len(devices))
+			var input string
+			_, err := fmt.Scanln(&input)
+			if err != nil {
+				fmt.Printf("Error reading input: %v\n", err)
+				continue
+			}
+
+			// Verify we got an integer.
+			selection, err = strconv.Atoi(input)
+			if err != nil {
+				fmt.Printf("Invalid input '%s'\n", input)
+				continue
+			}
+
+			// Verify selection is within range.
+			if selection < 1 || selection > len(devices) {
+				fmt.Printf("Invalid MFA device selected\n")
+				continue
+			}
+			break
+		}
+
+		deviceID = fmt.Sprintf("%v", devices[selection-1].DeviceId)
+		deviceType = devices[selection-1].DeviceType
+
+	} else {
+		deviceID = fmt.Sprintf("%v", devices[0].DeviceId)
```

Comment:
This will panic when `len(devices) == 0`.

I suggest the following logic (assuming the previous changes are already implemented):

```go
if len(d) == 0 {
    return nil, errors.New("empty device list")
}

if len(d) == 1 {
    return d[0], nil
}

// Continue with selecting a device etc.
```

The Golang best practice here is to handle special cases first and return ASAP instead of using `else`. This keeps the code clean, readable and less indented.